build.go

```diff
@@ -155,7 +176,7 @@ func Build(entryResolver EntryResolver,
 					layer.BuildEnv.Default("NPM_CONFIG_GLOBALCONFIG", globalNpmrcPath)
 				}
 				path := filepath.Join(layer.Path, "node_modules", ".bin")
-				layer.BuildEnv.Append("PATH", path, string(os.PathListSeparator))
+				layer.BuildEnv.Prepend("PATH", path, string(os.PathListSeparator))
```

Comment:
I'm still wondering a bit about this part of the change. With the append, installed packages could add a binary but not replace one of the system ones. With Prepend they can replace system ones which might be a security concern.  

I assume it would be possible to pre-pend just npm, with something like

path := filepath.Join(layer.Path, "node_modules", ".bin")
layer.BuildEnv.Prepend("PATH", filepath.Join(path, "npm"), path, string(os.PathListSeparator))
layer.BuildEnv.Append("PATH", path, string(os.PathListSeparator))

which would still let us use the newer npm, but not change anything else?

Reply:
Sounds good. I will adapt it.

Reply:
> which would still let us use the newer npm, but not change anything else

I adapted the code to use a combination of `prepend` and `append`, but now many integration test fail and it seems that only the `prepend` is taking affect. Not the combination.

Reply:
> which would still let us use the newer npm, but not change anything else

I did the change like proposed, but now many integration test fail. Only `prepend` seem to be used and not the combination of `prepend` and `append`. Is there a reason why not both can be used at the same time?

Reply:
Can you post the code you tried?  I don't see any reason why it should not be possible to do both

Reply:
Proposed fix: https://github.com/paketo-buildpacks/packit/pull/581

Reply:
> I'm still wondering a bit about this part of the change. With the append, installed packages could add a binary but not replace one of the system ones. With Prepend they can replace system ones which might be a security concern.
> 
> I assume it would be possible to pre-pend just npm, with something like
> 
> path := filepath.Join(layer.Path, "node_modules", ".bin") layer.BuildEnv.Prepend("PATH", filepath.Join(path, "npm"), path, string(os.PathListSeparator)) layer.BuildEnv.Append("PATH", path, string(os.PathListSeparator))
> 
> which would still let us use the newer npm, but not change anything else?

We ran into a problem with that one. It worked for `ubi`, but the shell of the paketo-builder doesn't like binaries in the `PATH`, only directories. So we created a folder containing only a symlink to `npm` (unfortunately one more symlink) and prepended this folder to `PATH`. WDYT?